File: POLIMORFISMO/EJERCICIO-1/Videojuego.java
public class Videojuego {
    private String nombre;
    private String plataforma;
    private int cantidadJugadores;

    public Videojuego(String nombre, String plataforma, int cantidadJugadores) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.cantidadJugadores = cantidadJugadores;
    }

    public Videojuego(String nombre, String plataforma) {
        this(nombre, plataforma, 1);
    }
    public void mostrar() {
        System.out.println("Nombre: " + nombre);
        System.out.println("Plataforma: " + plataforma);
        System.out.println("Cantidad de jugadores: " + cantidadJugadores);
    }
    public void agregarJugadores() {
        this.cantidadJugadores += 1;
    }
    public void agregarJugadores(int cantidad) {
        this.cantidadJugadores += cantidad;
    }
    public static void main(String[] args) {
        Videojuego juego1 = new Videojuego("Dragon Ball", "PlayStation", 2);
        Videojuego juego2 = new Videojuego("Left4Dead", "PC");  
    
        System.out.println("== Videojuego 1 ==");
        juego1.mostrar();

        System.out.println("\n== Videojuego 2 ==");
        juego2.mostrar();

        juego1.agregarJugadores();       
        juego2.agregarJugadores(3);      

        System.out.println("\n== Después de agregar jugadores ==");
        System.out.println("\n== Videojuego 1 ==");
        juego1.mostrar();

        System.out.println("\n== Videojuego 2 ==");
        juego2.mostrar();
    }
}

File: POLIMORFISMO/EJERCICIO-1/Videojuego.py
class Videojuego:
    def __init__(self, nombre, plataforma, cantidadJugadores):
        self.nombre = nombre
        self.plataforma = plataforma
        self.cantidadJugadores = cantidadJugadores
    def mostrar(self):
        print(f"Nombre: {self.nombre}")
        print(f"Plataforma: {self.plataforma}")
        print(f"Cantidad de jugadores: {self.cantidadJugadores}")
    def agregarJugadores(self, cantidad=1):
        self.cantidadJugadores += cantidad

juego1 = Videojuego("Dragon Ball", "PlayStation", 2)
juego2 = Videojuego("Left4Dead", "PC", 1)

juego1.mostrar()
print()
juego2.mostrar()

juego1.agregarJugadores()      
juego2.agregarJugadores(3)      

print("\nDespués de agregar jugadores:")
juego1.mostrar()
print()
juego2.mostrar()
